Support for Go 1.25

During the testing of Go 1.25rc1 in
* https://github.com/Homebrew/homebrew-core/pull/226636

it was noted that the build of Ignite CLI [fails](https://github.com/Homebrew/homebrew-core/actions/runs/15601592250/job/43943411977?pr=226636#step:3:12637) with:
```
# github.com/bytedance/sonic/internal/rt
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/bytedance/sonic@v1.13.2/internal/rt/stubs.go:33:22: undefined: GoMapIterator
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/bytedance/sonic@v1.13.2/internal/rt/stubs.go:36:54: undefined: GoMapIterator
```

The upstream module `github.com/bytedance/sonic` already has a fix 
* https://github.com/bytedance/sonic/pull/808

merged and available in the [v1.14.0-rc1](https://github.com/bytedance/sonic/releases/tag/v1.14.0-rc1) prerelease. Initial testing shows that updating that module fixes the build.

Please
* `go get -u github.com/bytedance/sonic@v1.14.0-rc1`
* run the CI tests to make sure nothing is broken
* test with Go 1.25rc1 and fix any other changes needed and/or report bugs upstream
* wait until 1.14.0 or later [release](https://github.com/bytedance/sonic/releases) is available
* update the `github.com/bytedance/sonic` module (eg via `go get -u github.com/bytedance/sonic@latest`)
* create a release

Please note that [Go 1.25rc2 is expected on Tuesday](https://groups.google.com/g/golang-announce/c/q2mjAt9PfF4/m/Tpt8JVViAwAJ) and final is scheduled for August. It would be ideal if you make a release with Go 1.25 support by then or immediately after the release.

--- a/go.mod
+++ b/go.mod
@@ -133,8 +133,8 @@ require (
 	github.com/bufbuild/protovalidate-go v0.8.2 // indirect
 	github.com/butuzov/ireturn v0.3.1 // indirect
 	github.com/butuzov/mirror v1.3.0 // indirect
-	github.com/bytedance/sonic v1.13.2 // indirect
-	github.com/bytedance/sonic/loader v0.2.4 // indirect
+	github.com/bytedance/sonic v1.14.0 // indirect
+	github.com/bytedance/sonic/loader v0.3.0 // indirect
 	github.com/calmh/randomart v1.1.0 // indirect
 	github.com/catenacyber/perfsprint v0.8.1 // indirect
 	github.com/ccojocar/zxcvbn-go v1.0.2 // indirect

